htmd/qm/base.py

```diff
@@ -160,13 +160,11 @@ def _submit(self):
         for directory in self._directories:
             if not self._completed(directory):
                 self.queue.submit(directory)
+        self.queue._dirs = self._directories
```

Comment:
It is not the best idea to modify a private attribute out side of the object.

Reply:
I'm willing to make that compromise in order for people who run with Lsf or Slurm being able to restart a parameterize job, which is not possible currently

Reply:
I don't get how this helps. Isn't this make to restart all simulation, even the ones, which already finished successfully?

It just copy `_directories` to `_dirs`, so it not checked with `_completed` method. Or am I missing something?

Reply:
It will not matter, as the completed ones should also have sentinel files.
By copying the `_directories` into `_dirs`, the `notcompleted` method actually works instead of failing due to having the `_dirs` being empty (which is a RuntimeError in general runs, but not on a restart run). We never devised the queues to manage the restart of simulations (MD or QM). We may want to make it more general and implemented in that way, but right now we need a fix

Reply:
I see. If you just remove `if not self._completed(directory)`, all directories would be added unconditionally by `submit`. And this will avoid modifying the private variable directly.

Reply:
Ok, yes, it's true. The problem is that submitting rewrites inputs, cleans sentinels and will actually launch the jobs again, not caring about the output that is already there.